src/Orleans.Core.Abstractions/Streams/Core/IAsyncBatchProducer.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams
+{
+    /// <summary>
+    /// This interface generalizes the IAsyncObserver interface to allow production of batches of items.
+    /// <para>
+    /// Note that this interface is invoked (used) by item producers.
+    /// </para>
+    /// </summary>
+    /// <typeparam name="T">The type of object consumed by the observer.</typeparam>
+    public interface IAsyncBatchProducer<T> : IAsyncObserver<T>
```

Comment:
The naming here confused me - I feel `IAsyncBatchObserver` would be more obvious. I see that this interface is implemented by the producing side of the stream and called by the producer, but it feels inconsistent with the naming of the base interface, `IAsyncObserver`

Reply:
There is already an IAsyncBatchObserver used in the consumer side.  This is what I meant by the surfaces are asymmetric. :/